react-design/my-app/src/RotatingBanner.js

```diff
@@ -0,0 +1,44 @@
+import BannerButton from './BannerButton';
+import BannerTitle from './BannerTitle';
+import './RotatingBanner.css';
+import { useState } from 'react';
+
+export default function RotatingBanner({ itemsArray }) {
+  const [bannerIndex, setBannerIndex] = useState(0);
+
+  function increment() {
+    // used with the 'Next' button
+    setBannerIndex((bannerIndex + 1) % itemsArray.length);
+  }
+  function decrement() {
+    // used with the 'Prev' button
+    setBannerIndex(customModulo(bannerIndex - 1, itemsArray.length));
+  }
+  function customIndex(text) {
+    // used with the number buttons
+    setBannerIndex(text);
+  }
+  const numberButtons = itemsArray.map((element, index) => (
+    <BannerButton
+      key={index}
+      text={index}
+      selected={index === bannerIndex}
+      onCustomClick={customIndex}></BannerButton>
+  ));
+
+  return (
+    <div className="banner">
+      <BannerTitle text={itemsArray[bannerIndex]}></BannerTitle>
+      <div className="buttons">
+        <BannerButton text="Prev" onCustomClick={decrement}></BannerButton>
+        <div className="num-buttons">{numberButtons}</div>
+        <BannerButton text="Next" onCustomClick={increment}></BannerButton>
+      </div>
+    </div>
+  );
+}
+
+function customModulo(n, m) {
+  // used because Javascript doesn't handle negative modulo 'correctly'.
```

Comment:
I wouldn't say it's necessarily "incorrect" (as JS does calculate the modulo). It's just that "modulo" and "mathematical remainder" are two different things, and what you need is the mathematical remainder.